### DISCUSSION TITLE
[Combobox] Keypress control when Options not open.

### DISCUSSION THREAD

**Discussion by FranciscoMessina** [tailwindlabs/headlessui#1198] [Ideas]
Hello, this is my first time writing here, so I apologize in advance in case I am doing something wrong. I was looking for a way to make a autocomplete with typeahead in an textarea (Twitter mentions style), and came across the Combobox component. I am building a input where you can search for elements (like products or quick answers) and quickly insert the url to the product or the quick answer in it. I managed to use the Combobox to do it, but I had a onKeypress event listener so when the user pressed CRTL + Enter the input was submitted. However the combobox has a event listener that prevents default and stops propagation on the Enter key, when you have options showing this is completely fine, but if there are no option shown and the Options is closed, I think it would be nice to be hable to use all the Keys. The way I thought to workaround this is adding a isOpen prop and if it is not open then ignore the default event listener and continue to the custom one, in this case crtl + enter.  I hope what I wrote is understandable. 

I leave some images of the component I am building, not sure if they help or not.
![image](https://user-images.githubusercontent.com/91616369/156902616-16a147e5-286c-4a81-9e5b-721e92edc028.png)
![image](https://user-images.githubusercontent.com/91616369/156902746-c70161ce-88f9-414f-83ff-c56d023b18fe.png)
![image](https://user-images.githubusercontent.com/91616369/156902748-388f2375-3552-4e13-b3e8-9cb23d6254db.png)

**Comment by jamesthomsondev** [comment]
@FranciscoMessina I think your issue is actually that modifiers aren't supported on the `keyPress` event.

>The keypress event is fired when a key that produces a character value is pressed down.

>Examples of keys that produce a character value are alphabetic, numeric, and punctuation keys. Examples of keys that don't produce a character value are modifier keys such as Alt, Shift, Ctrl, or Meta.

Additionally, `onKeyPress` has [been deprecated](https://developer.mozilla.org/en-US/docs/Web/API/Document/keypress_event). I

Instead you should use `onKeyDown` or `onKeyUp`. That said, `onKeyDown` seems to swallow the event so as a workaround I had to use `onKeyUp`. Hope that helps!

**Reply by FranciscoMessina** [reply]
Hi! Thank you a lot for the answer! I wasn't aware that it was deprecated, will change to use another event as soon as possible, however the functionality was working fine when not using the combobox, and making a small change in the source code of it made it work again.
